.github/workflows/linux_meson.yml

```diff
@@ -178,3 +173,30 @@ jobs:
         python -c "import scipy"
         python -c "import scipy.linalg"
         python -c "from scipy import cluster; cluster.test()"
+
+  python_debug:
+    name: Python-debug & ATLAS
+    if: "github.repository == 'scipy/scipy' || github.repository == ''"
+    runs-on: ubuntu-22.04  # provides python3.10-dbg
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Configuring Test Environment
+        run: |
+          sudo apt-get update
+          sudo apt install python3-dbg python3-dev libatlas-base-dev liblapack-dev gfortran ccache libgmp-dev libmpfr-dev libmpc-dev
+          python3-dbg --version # just to check
+          python3-dbg -c 'import sys; print("Python debug build:", hasattr(sys, "gettotalrefcount"))'
+      - name: Build SciPy
+        run: |
+          python3-dbg -m pip install build
+          python3-dbg -m build --wheel -Csetup-args=-Dbuildtype=debugoptimized -Csetup-args=-Dblas=blas-atlas -Csetup-args=-Dlapack=lapack-atlas -Ccompile-args=-j2
+          python3-dbg -m pip install dist/scipy*.whl
+      - name: Testing SciPy
+        run: |
+          cd doc
+          python3-dbg -m pip install pytest pytest-xdist pytest-timeout mpmath gmpy2 threadpoolctl pooch
+          python3-dbg -m pytest --pyargs scipy -n2 --durations=10 -m "not slow"
```

Comment:
No particular reason beyond this seems convenient and pretty standard to work with an installed `scipy` package rather than one that's also built locally by `dev.py`.